### PR TITLE
Deriving public key from private key

### LINKED ISSUE #60: Public key can be derived from private one, so we don't have to ask for public key, where we need private one anyway.

### PR DESCRIPTION
**Pull request by gabrielfior**
Closes #60 

- Loads private_key from the ENV variables (as done previously) and derives public key
- Added test

## Summary by CodeRabbit

- **New Features**
	- Enhanced configuration with new environment variables for improved security.
	- Added functionality to convert private keys to public keys for address verification.

- **Refactor**
	- Improved secret handling in deployment and market replication scripts.
	- Enhanced security by deriving `bet_from_address` directly from the private key.

- **Tests**
	- Ensured reliability with new tests for the `private_key_to_public_key` function.

### CODE REVIEW

**Review by evangriffiths — COMMENTED** [review]
Looks good, but I would say that to close https://github.com/gnosis/prediction-market-agent-tooling/issues/60 there are a bunch more places where `bet_from_address` is passed in along side a private key. Most cases are in markets/omen/omen.py

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,6 @@
+MANIFOLD_API_KEYS=[]
+OMEN_PUBLIC_KEYS=[]
+LOAD_FROM_GCP=true
+BET_FROM_ADDRESS=
+BET_FROM_PRIVATE_KEY=
+OPENAI_API_KEY=
```
Comment (kongzii) [inline .env.example:6]: We have two environmental configs:

1. APIKeys loaded from `.env` https://github.com/gnosis/prediction-market-agent-tooling/blob/0ed0dca0dab9eac375b0682e07e5c218152a0244/prediction_market_agent_tooling/config.py#L20

with

```
MANIFOLD_API_KEY
BET_FROM_ADDRESS
BET_FROM_PRIVATE_KEY
OPENAI_API_KEY
```

and

2. MonitorSettings loaded from `.env.monitor`: https://github.com/gnosis/prediction-market-agent-tooling/blob/c15f9e62b0567bd1dbd8f6476fe969a11d2d20f8/prediction_market_agent_tooling/monitor/monitor.py#L29

with

```
LOAD_FROM_GCP
MANIFOLD_API_KEYS
OMEN_PUBLIC_KEYS
PAST_N_WEEKS
```

---

It's super to have an example for newcomers but this one mixes them together, could you split it into two files `.env.example` and `.env.monitor.example`?

Also thanks to this PR, you can remove `BET_FROM_ADDRESS` from it 🎉

**Comment by kongzii** [comment]
Coool!

File: .env.example [hunk 2]
```diff
@@ -1,6 +1,3 @@
-MANIFOLD_API_KEYS=[]
-OMEN_PUBLIC_KEYS=[]
-LOAD_FROM_GCP=true
-BET_FROM_ADDRESS=
+MANIFOLD_API_KEY=[]
```
Comment (kongzii) [inline .env.example:1]: `MANIFOLD_API_KEY` should be a string, not a list

File: examples/cloud_deployment/gcp/deploy.py [hunk 3]
```diff
@@ -53,9 +52,6 @@ def main(
         secrets=json.loads(secrets) if secrets else None,
         memory=512,
         api_keys=APIKeys(
-            BET_FROM_ADDRESS=(
-                verify_address(bet_from_address) if bet_from_address else None
-            ),
```
[the remark this answers is not in the document]
Comment (kongzii) [reply examples/cloud_deployment/gcp/deploy.py:58]: @gabrielfior I think Evan meant `bet_from_address: str | None = None,` in this function definition

File: examples/replicate_markets/deploy.py [hunk 4]
```diff
@@ -30,7 +29,6 @@ def main(
     More manual example for cases where the default behavior falls short.
     """
     api_keys = APIKeys(
-        BET_FROM_ADDRESS=verify_address(from_address),
```
Comment (kongzii) [inline examples/replicate_markets/deploy.py:33]: In this one as well 🙏 

Screenshot by Dropbox Capture

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 5]
```diff
@@ -393,7 +386,7 @@ def omen_sell_outcome_tx(
     Sells the given amount of shares for the given outcome in the given market.
     """
     amount_wei = xdai_to_wei(amount)
-    from_address_checksummed = Web3.to_checksum_address(from_address)
+    Web3.to_checksum_address(from_address)
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/markets/omen/omen.py:389]: Agreed with Rabbit, this line is now unused and `from_address: ChecksumAddress,` can be removed from the function arguments

File: examples/replicate_markets/deploy.py (hunk 4 above)
Comment (gabrielfior) [reply examples/replicate_markets/deploy.py:33]: Done in https://github.com/gnosis/prediction-market-agent-tooling/pull/113

**Pull request merged**